app/src/main/java/com/sdp13epfl2021/projmag/database/interfaces/UserdataDatabase.kt

```diff
@@ -152,4 +153,26 @@ interface UserdataDatabase {
         onSuccess: () -> Unit,
         onFailure: (Exception) -> Unit
     )
+
+    /**
+     * Uploads the user profile to a database
+     * @param profile the user profile to upload
+     * @param onSuccess the closure that's when a profile is uploaded successfully
+     * @param onFailure the closure that's called if the upload fails
+     */
+    fun uploadProfile(
+        profile: ImmutableProfile,
+        onSuccess: () -> Unit,
+        onFailure: (Exception) -> Unit
+    )
+
+    /**
+     * Gets the user profile from the database if it exists
+     * @param onSuccess the closure that's when a profile is downloaded successfully with the fetched profile passed to it
+     * @param onFailure the closure that's called if the download fails
+     */
+    fun getProfile(
+        onSuccess: (profile: ImmutableProfile?) -> Unit,
+        onFailure: (Exception) -> Unit
+    )
```

Comment:
If we can't specify userID, we can't get profile of other users.
Specify name for lambda is not necessary.
Somethings like this will be better.
```
    fun getProfile(
        userID: String,
        onSuccess: (ImmutableProfile?) -> Unit,
        onFailure: (Exception) -> Unit
    )
````